onnx/large_helper.py

```diff
@@ -0,0 +1,333 @@
+# Copyright (c) ONNX Project Contributors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import enum
+import os
+from typing import Any, Dict, Iterable, Optional, Set, Tuple
+
+import numpy as np
+
+from onnx import (
+    GraphProto,
+    ModelProto,
+    StringStringEntryProto,
+    TensorProto,
+    checker,
+    load_model,
+)
+from onnx.external_data_helper import (
+    ExternalDataInfo,
+    _get_all_tensors,
+    _sanitize_path,
+    uses_external_data,
+)
+from onnx.helper import make_model
+
+
+def make_large_tensor_proto(
+    location: str, name: str, tensor_type: int, shape: Tuple[int, ...]
+) -> TensorProto:
+    """
+    Create an external tensor.
+
+    :param location: unique identifier (not necessary a path)
+    :param name: initializer name in the graph
+    :param tensor_type: onnx type
+    :param shape: shape the of the initializer
+    :return: the created tensor
+    """
+    tensor_location = location
+    tensor = TensorProto()
+    tensor.name = name
+
+    del tensor.external_data[:]
+    tensor.data_location = TensorProto.EXTERNAL
+    for k, v in {
+        "location": tensor_location,
+        "offset": None,
+        "length": None,
+        "checksum": None,
+        "basepath": None,
+    }.items():
+        if v is not None:
+            entry = tensor.external_data.add()
+            entry.key = k
+            entry.value = str(v)
+
+    tensor.data_type = tensor_type
+    tensor.dims.extend(shape)
+    return tensor
+
+
+class LargeModelFileFormat(enum.IntEnum):
+    # One file for all the weights.
+    SINGLE_TENSOR_FILE = 2
```

Comment:
Is this in sync with any proto definition? Curious why it starts from 2?

Reply:
Why not use a boolean `all_tensors_to_one_file: bool` just like some of the functions? If we plan to introduce newer file-formats later, we could introduce these generalizations then.

Reply:
removed